modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RetryingInputStream.java

```diff
@@ -166,6 +166,12 @@ private void ensureOpen() {
     }
 
     private void reopenStreamOrFail(IOException e) throws IOException {
+        if (purpose == OperationPurpose.REPOSITORY_ANALYSIS) {
+            logger.warn(() -> format("""
+                failed reading [%s/%s] at offset [%s]""", blobStore.bucket(), blobKey, start + currentOffset), e);
+            throw e;
+        }
```

Comment:
I wonder whether we should add a TODO here to remind this needs to be adjusted when we upgrade to v2 client?

Reply:
I'm not sure we'd notice such a TODO without actually seeing the test fail. Or put differently I think there'll be quite a few other things like this which need adjusting if we upgrade to SDKv2.